CMakeLists.txt

```diff
@@ -91,6 +91,8 @@ endif()
 
 include_directories(${CMAKE_BINARY_DIR}) # Make the exv_conf.h file visible for the full project
 
+find_package(fmt REQUIRED)
```

Comment:
Gate by C++20?

Reply:
std::format is only supported by MSVC unfortunately

Reply:
https://stackoverflow.com/a/69156536

Reply:
So a check for C++20 level and `format` via https://cmake.org/cmake/help/latest/module/CheckIncludeFileCXX.html (instead of specific compiler) should suffice.

Reply:
Those compiler versions are way too recent. I wouldn’t want to kill compatibility like that.

Reply:
I didn't say kill, just don't depend on fmtlib when you don't have to. Yes, it means ugly #ifdefs in code, but I feel it's good to think ahead here and make phasing out fmtlib in the future easy.

Reply:
That makes 0 sense. fmt and printf formats are different.

Reply:
I'm afraid there's still a misunderstanding...

I'm suggesting to code straight for C++20 std::format and use fmtlib as a fallback only in the case the former is not yet available.

Something like
```
#if __has_include(<format>)
#include <format>
#else
#include <fmt/core.h>
#define std::format fmt::format
```

This should work for most "regular" cases - I haven't checked if you used for chrono etc.

Then you'd also need to handle the detection in CMake and Meson to depend on fmtlib only when necessary, as said.

Reply:
```
/home/mangix/devstuff/exiv2/src/webpimage.cpp:443:48: error: ‘format’ is not a member of ‘std’; did you mean ‘fmt::v9::format’?
  443 |         out << Internal::indent(depth) << std::format("  {} | {:8} | {:8} | ", chunkId.c_str(), size, offset)
      |                                                ^~~~~~
In file included from /home/mangix/devstuff/exiv2/src/image_int.hpp:15,
                 from /home/mangix/devstuff/exiv2/src/webpimage.cpp:16:
/usr/include/fmt/core.h:3204:31: note: ‘fmt::v9::format’ declared here
 3204 | FMT_NODISCARD FMT_INLINE auto format(format_string<T...> fmt, T&&... args)
      |                               ^~~~~~

```

Reply:
GCC ifdefs out most of the <format> header if you don't have -std=c++20 I think. Anyhow, we should be able to make this work somehow...